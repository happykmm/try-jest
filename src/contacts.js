import _ from 'lodash';

const contacts = [
  {
    name: 'li qiang',
    phone: '13500001112',
  },
  {
    name: 'zhang qiang',
    phone: '13600001111',
  },
  {
    name: 'li gang',
    phone: '13700001111',
  }
];

export const searchContact = (term) => {
  return _.filter(contacts, c => _.includes(c.name, term));
}
