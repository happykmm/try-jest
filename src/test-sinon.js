const sinon = require('sinon');

const spy = sinon.spy();

console.log(spy())