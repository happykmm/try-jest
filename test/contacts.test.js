import _ from 'lodash';
import { searchContact } from '../src/contacts';

expect.addSnapshotSerializer({
  test: (val) => {
    return _.isArray(val);
  },
  print: (val) => {
    return _.chain(val)
      .map(item => `${item.name} - ${item.phone}`)
      .join('\n')
      .value();
  },
});

it('search ang', () => {
  expect(searchContact('ang')).toMatchSnapshot();
});

