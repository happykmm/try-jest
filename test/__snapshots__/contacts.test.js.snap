// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search ang 1`] = `
li qiang - 13500001112
zhang qiang - 13600001111
li gang - 13700001111
`;
